modules/_labs/core/react/lib/CanvasProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import {ThemeProvider} from 'emotion-theming';
+import {InputProvider} from '@workday/canvas-kit-react-core';
+import {CanvasTheme} from './theming/types';
+import {defaultCanvasTheme} from './theming/theme';
+
+export interface CanvasProviderProps {
+  theme: CanvasTheme;
+  setThemeGlobal: boolean;
```

Comment:
`setThemeGlobal` seems to encapsulate the concerns of two different components here. To me a 'provider' provides the subtree with some data that it needs to render properly (in this case the theme). But with `setThemeGlobal`, this component also can also modify the window which is a global side-effect.

Because of this, we can have some weird issues when `setThemeGlobal` is true on multiple providers. The global theme can flip-flop causing places that don't use a `CanvasProvider` for their theme to unexpectedly change.

Could we provide a separate component to explicitly take the role of setting the global instead? That way we're declaratively doing a side-effect with that one and the `CanvasProvider` can remain free from it (pure). Alternatively, we could just document that this component comes with a side-effect that users need to be aware of but my gut feeling is that this isn't a typical React pattern to have a component do two very distinct things b/c of the unpredictable behavior if used incorrectly.

Usage:
```jsx
// In the main App.jsx
// Sets global theme and other global things to your app maybe
<CanvasGlobal theme={...} />

// Somewhere else, I need a more specific theme for this subtree
<CanvasProvider theme={...}>
...
</CanvasProvider>
```

Reply:
Totally agree - good call. I will revise.

Reply:
@lychyi I just pushed a change for this. I ended up just [recommending that they add the theme to the window object themselves](https://github.com/Workday/canvas-kit/pull/272/commits/a4f0fa79ac3404a7d9a6facc39f6c754ebea24c3#diff-c895718dba9f9b5b971bb7cf5538a92fR167). It's easier for non-react apps to do this where needed. It seemed overkill to have a component just to do this. Let me know your thoughts.

Reply:
@anicholls I think that's a good idea. We can always think about it later if people are looking for a declarative way to handle theming on the window.